src/util/audio/AudioPlayer.cpp

```diff
@@ -25,44 +26,78 @@ AudioPlayer::~AudioPlayer()
 	XOJ_RELEASE_TYPE(AudioPlayer);
 }
 
-void AudioPlayer::start(string filename, unsigned int timestamp)
+bool AudioPlayer::start(string filename, unsigned int timestamp)
 {
 	XOJ_CHECK_TYPE(AudioPlayer);
 
 	// Start the producer for reading the data
-	this->vorbisProducer->start(std::move(filename), this->portAudioConsumer->getSelectedOutputDevice(), timestamp);
-	SF_INFO* vi = this->vorbisProducer->getSignalInformation();
+	bool status = this->vorbisProducer->start(std::move(filename), timestamp);
 
 	// Start playing
-	this->portAudioConsumer->startPlaying(static_cast<double>(vi->samplerate), static_cast<unsigned int>(vi->channels));
-
-	// Clean up after audio is played
-	stopThread = std::thread([&]
-							 {
-								 while (this->portAudioConsumer->isPlaying())
-								 {
-									 Pa_Sleep(100);
-								 }
-								 this->stop();
-							 });
-	stopThread.detach();
+	if (status)
+	{
+		status &= this->play();
```

Comment:
This probably should be logical or? Even if this probably works in 99% of all the cases with bool....